packages/flutter/lib/src/rendering/table_border.dart

```diff
@@ -194,49 +190,49 @@ class TableBorder {
     assert(canvas != null);
     assert(rect != null);
     assert(rows != null);
-    assert(rows.length >= 2);
-    assert(rows.first == 0.0);
-    assert(rows.last == rect.height);
+    assert(rows.isEmpty || (rows.first > 0.0 && rows.last < rect.height));
     assert(columns != null);
-    assert(columns.isNotEmpty);
-    assert(columns.first == 0.0);
-    assert(columns.last < rect.width);
+    assert(columns.isEmpty || (columns.first > 0.0 && columns.last < rect.width));
 
     final Paint paint = new Paint();
     final Path path = new Path();
 
-    switch (verticalInside.style) {
-      case BorderStyle.solid:
-        paint
-          ..color = verticalInside.color
-          ..strokeWidth = verticalInside.width
-          ..style = PaintingStyle.stroke;
-        path.reset();
-        for (int x = 1; x < columns.length; x += 1) {
-          path.moveTo(rect.left + columns[x], rect.top);
-          path.lineTo(rect.left + columns[x], rect.bottom);
-        }
-        canvas.drawPath(path, paint);
-        break;
-      case BorderStyle.none:
-        break;
+    if (columns.isNotEmpty) {
```

Comment:
similarly here, checking isNotEmpty means you walk two steps into the original list (first looking at the last entry, then the second to last entry, via multiple objects, the original list, the reversed iterator, and the skip iterator) before you can determine if it's empty.

Reply:
I can remove this check.  We'll just end up drawing an empty path (which is what we used to do anyway).

Reply:
I think it's fine with the check, I just wanted to make sure the implications of using an Iterable (vs reifying the list) were clear.

Reply:
I'm going to land it without the checks.  It's conceptually a separate change (a performance optimization rather than an API cleanup), and likely a premature optimization at that.